python-api/gstswitch/controller.py

```diff
@@ -394,3 +424,57 @@ def parse_preview_ports(cls, res):
         for tupl in liststr:
             preview_ports.append(int(tupl[0]))
         return preview_ports
+
+    def on_preview_port_added(self, callback):
+        """Register a Callback for the preview_port_added Signal
```

Comment:
No, that is the `serve` parameter, it can either be `GST_SERVE_VIDEO_STREAM` or `GST_SERVE_VIDEO_AUDIO` (`GstSwitchServeStreamType`)

The `type` Parameter is of Type `GstCaseType` which seems to be completely internal…